# RPATH issues when upgrading net-snmp to version 5.4.1
#
# Developed in-house. Solaris-specific; not suitable for upstream

diff -wpruN '--exclude=*.orig' net-snmp-5.8~/ltmain.sh net-snmp-5.8/ltmain.sh
--- net-snmp-5.8~/ltmain.sh	2018-07-16 14:33:40.000000000 +0000
+++ net-snmp-5.8/ltmain.sh	2018-07-24 09:51:35.687556392 +0000
@@ -5406,7 +5406,9 @@ func_mode_link ()
 	  if test "$prev" = rpath; then
 	    case "$rpath " in
 	    *" $arg "*) ;;
-	    *) func_append rpath " $arg" ;;
+	    *) func_append rpath " $arg"
+		# This is make sure only $libdir path is encoded in RUNPATH
+			sma_rpath="$rpath" ;;
 	    esac
 	  else
 	    case "$xrpath " in
@@ -8046,7 +8048,7 @@ EOF
 		fi
 	      else
 		eval flag=\"$hardcode_libdir_flag_spec\"
-		func_append dep_rpath " $flag"
+#func_append dep_rpath " $flag"
 	      fi
 	    elif test -n "$runpath_var"; then
 	      case "$perm_rpath " in
